ocaml/libs/xapi-compression/xapi_compression.ml

```diff
@@ -45,10 +51,10 @@ module Make (Algorithm : ALGORITHM) = struct
     Unixfd.with_pipe ~loc:__LOC__ () @@ fun zcat_out zcat_in ->
     let args =
       match mode with
-      | Compress ->
-          []
-      | Decompress ->
-          ["--decompress"; "--stdout"]
+      | Compress args ->
+          List.concat [Algorithm.compress_options; args]
```

Comment:
The args here would probably have to be mapped to the underlying compression algo somehow, each tool has its own args, perhaps we could have a variant here and each implementation can map those variants to string args?
There are at least 2 uses of compression so far:
* compress once, use multiple times (.xva), spending  bit more time on compression could be worth it if you save on bandwidth and make downloads faster
* compress/decompress once, during live migrate: balance CPU usage with network bandwidth

So perhaps to start 2 variants:
`type compression_option = BalanceRatio | BalanceTransferSpeed`

For decompression we have no options yet so that could be just the `unit` type for now.
Then gzip and zstd could map these.
E.g. zstd could map it like this:
```
function
| BalanceTransferSpeed -> ["--adapt=min=1,max=3"]
| BalanceRatio -> [] (* we could specify some higher level here, but lets leave as is for now to avoid making it too slow *)
```

And gzip:
```
function
| BalanceTransferSpeed ->  ["--fast"]
| BalanceRatio -> [] (* could specify --best but gzip is too slow already... *)
```

Reply:
The `Algorithm.compress_options` and `Algorithm.decompress_options` cover the default per algorithm. It is a remaining limitation that any additional options are just added to these and you can't replace the default. I am reluctant to add more abstractions at this point before it is clear what actual use cases will look like.